Test failure in ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py.TestUpgradeThenRollback.test_setup_in_blob_storage_config[restart_prestable-26-1_to_prestable-26-2]

ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py.TestUpgradeThenRollback.test_setup_in_blob_storage_config[restart_prestable-26-1_to_prestable-26-2]

**Test Failure**

**Build Information:**
- Version: 5b365c8c66bb6df0b1f822d21a5fa92f5943e9b2
- Build Preset: relwithdebinfo
- Branch: main

**Test Report:** [View report](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21834015376/ya-main-x86-64/try_1/ya-test.html#FAIL)

**Run Command:**
```
./ya make -ttt --build "relwithdebinfo" -F 'test_infer_pdisk_expected_slot_count.py::TestUpgradeThenRollback::test_setup_in_blob_storage_config[restart_prestable-26-1_to_prestable-26-2]' ydb/tests/compatibility
```

**Owner:** [TEAM:@ydb-platform/engineering](https://github.com/orgs/ydb-platform/teams/engineering)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=main&full_name=ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py.TestUpgradeThenRollback.test_setup_in_blob_storage_config[restart_prestable-26-1_to_prestable-26-2]&build_type=relwithdebinfo)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [log](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21834015376/ya-main-x86-64/try_1/artifacts/logs/ydb/tests/compatibility/test-results/py3test/testing_out_stuff/test_infer_pdisk_expected_slot_count.py.TestUpgradeThenRollback.test_setup_in_blob_storage_config.restart_prestable-26-1_to_prestable-26-2.log)
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21834015376/ya-main-x86-64/try_1/test_artifacts/ydb-tests-compatibility.zip)

**Error Summary:**
```
ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py:276: in test_setup_in_blob_storage_config
    retry_assertions(check_pdisks, timeout)
ydb/tests/library/common/wait_for.py:94: in retry_assertions
    raise e
ydb/tests/library/common/wait_for.py:90: in retry_assertions
    return callable_()
ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py:272: in check_pdisks
    assert pdisk.PDiskMetrics.HasField('SlotCount') and pdisk.PDiskMetrics.SlotCount == CONST_INITIAL_SLOT_COUNT
E   assert (True and 24 == 14)
E    +  where True = <built-in method HasField of TPDiskMetrics object at 0x7fea295afad0>('SlotCount')
E    +    where <built-in method HasField of TPDiskMetrics object at 0x7fea295afad0> = AvailableSize: 515236691968\nTotalSize: 515396075520\nMaxReadThroughput: 127000000\nMaxWriteThroughput: 127000000\nNonReal... 125\nState: Normal\nUpdateTimestamp: 1770675299263458\nSlotCount: 24\nSlotSizeInUnits: 2\nPDiskUsage: 0.006920415224913495\n.HasField
E    +      where AvailableSize: 515236691968\nTotalSize: 515396075520\nMaxReadThroughput: 127000000\nMaxWriteThroughput: 127000000\nNonReal... 125\nState: Normal\nUpdateTimestamp: 1770675299263458\nSlotCount: 24\nSlotSizeInUnits: 2\nPDiskUsage: 0.006920415224913495\n = NodeId: 1\nPDiskId: 1000\nPath: "SectorMap:TestInferPDiskSettings:480"\nPDiskConfig {\n  ExpectedSlotCount: 14\n}\nGuid: 145...\n  SlotSizeInUnits: 2\n  PDiskUsage: 0.006920415224913495\n}\nDecommitStatus: DECOMMIT_NONE\nMaintenanceStatus: NO_REQUEST\n.PDiskMetrics
E    +  and   24 = AvailableSize: 515236691968\nTotalSize: 515396075520\nMaxReadThroughput: 127000000\nMaxWriteThroughput: 127000000\nNonReal... 125\nState: Normal\nUpdateTimestamp: 1770675299263458\nSlotCount: 24\nSlotSizeInUnits: 2\nPDiskUsage: 0.006920415224913495\n.SlotCount
E    +    where AvailableSize: 515236691968\nTotalSize: 515396075520\nMaxReadThroughput: 127000000\nMaxWriteThroughput: 127000000\nNonReal... 125\nState: Normal\nUpdateTimestamp: 1770675299263458\nSlotCount: 24\nSlotSizeInUnits: 2\nPDiskUsage: 0.006920415224913495\n = NodeId: 1\nPDiskId: 1000\nPath: "SectorMap:TestInferPDiskSettings:480"\nPDiskConfig {\n  ExpectedSlotCount: 14\n}\nGuid: 145...\n  SlotSizeInUnits: 2\n  PDiskUsage: 0.006920415224913495\n}\nDecommitStatus: DECOMMIT_NONE\nMaintenanceStatus: NO_REQUEST\n.PDiskMetrics
```

--- a/ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py
+++ b/ydb/tests/compatibility/test_infer_pdisk_expected_slot_count.py
@@ -227,6 +227,8 @@ class TestUpgradeThenRollback(RestartToAnotherVersionFixture):
             }
         }
         self.replace_config(full_config)
+        inferred_slot_count = 24
+        inferred_slot_size_in_units = 2
         logger.info(f"Inferred PDisk setting applied {time.time()=}")
 
         self.cluster.client.pdisk_set_all_active(pdisk_path=CONST_PDISK_PATH)
@@ -241,8 +243,8 @@ class TestUpgradeThenRollback(RestartToAnotherVersionFixture):
                 assert pdisk.PDiskConfig.ExpectedSlotCount == CONST_INITIAL_SLOT_COUNT
                 assert pdisk.ExpectedSlotCount == CONST_INITIAL_SLOT_COUNT
                 assert pdisk.PDiskMetrics.TotalSize == CONST_480_GB
-                assert pdisk.PDiskMetrics.SlotCount == 24
-                assert pdisk.PDiskMetrics.SlotSizeInUnits == 2
+                assert pdisk.PDiskMetrics.SlotCount == inferred_slot_count
+                assert pdisk.PDiskMetrics.SlotSizeInUnits == inferred_slot_size_in_units
                 assert pdisk.PDiskMetrics.UpdateTimestamp * 1e-6 > t2
                 assert pdisk.PDiskMetrics.UpdateTimestamp * 1e-6 < deadline
         retry_assertions(check_pdisks, timeout)
@@ -268,9 +270,12 @@ class TestUpgradeThenRollback(RestartToAnotherVersionFixture):
                 if self.versions[0] < (25, 3):
                     assert not pdisk.PDiskMetrics.HasField('SlotCount')
                     assert not pdisk.PDiskMetrics.HasField('SlotSizeInUnits')
-                else:
+                elif self.versions[0] < (25, 5):
                     assert pdisk.PDiskMetrics.HasField('SlotCount') and pdisk.PDiskMetrics.SlotCount == CONST_INITIAL_SLOT_COUNT
                     assert pdisk.PDiskMetrics.HasField('SlotSizeInUnits') and pdisk.PDiskMetrics.SlotSizeInUnits == 0
+                else:
+                    assert pdisk.PDiskMetrics.SlotCount == inferred_slot_count
+                    assert pdisk.PDiskMetrics.SlotSizeInUnits == inferred_slot_size_in_units
                 assert pdisk.PDiskMetrics.UpdateTimestamp * 1e-6 > t3
                 assert pdisk.PDiskMetrics.UpdateTimestamp * 1e-6 < deadline
         retry_assertions(check_pdisks, timeout)

